mods/ra/maps/soviet-07/soviet07.lua

```diff
@@ -0,0 +1,346 @@
+RSoldierTrapTrigger = { CPos.New(72, 72), CPos.New(72,73), CPos.New(72,74) }
+SoldierTrap2Trigger = { CPos.New(51, 73), CPos.New(51, 74) }
+
+
+cameraCCTrigger = false
+Trigger.OnEnteredFootprint(CameraCCTrigger, function(a, id)
+	if not cameraCCTrigger and a.Owner == player then
+		cameraCCTrigger = true
+		Actor.Create("camera", true, { Owner = player, Location = CameraCC.Location })
+	end
+end)
+
+cameraGoalCenterTrigger = false
+Trigger.OnEnteredFootprint(CameraGoalCenterTrigger, function(a, id)
+	if not cameraGoalCenterTrigger and a.Owner == player then
+		cameraGoalCenterTrigger = true
+		if not controlCenterEngineerTrigger then
+			Actor.Create("camera", true, { Owner = player, Location = CameraGoalCenter1.Location })
+			Actor.Create("camera", true, { Owner = player, Location = CameraGoalCenter2.Location })
+			Actor.Create("camera", true, { Owner = player, Location = CameraGoalCenter3.Location })
+		end
+	end
+end)
+
+cameraGoalLeftTrigger = false
+Trigger.OnEnteredFootprint(CameraGoalLeftTrigger, function(a, id)
+	if not cameraGoalLeftTrigger and a.Owner == player then
+		cameraGoalLeftTrigger = true
+		Actor.Create("camera", true, { Owner = player, Location = CameraGoalLeft1.Location })
+		Actor.Create("camera", true, { Owner = player, Location = CameraGoalLeft2.Location })
+	end
+end)
+
+cameraGoalRightTrigger = false
+Trigger.OnEnteredFootprint(CameraGoalRightTrigger, function(a, id)
+	if not cameraGoalRightTrigger and a.Owner == player then
+		cameraGoalRightTrigger = true
+		Actor.Create("camera", true, { Owner = player, Location = CameraGoalRight1.Location })
+		Actor.Create("camera", true, { Owner = player, Location = CameraGoalRight2.Location })
+	end
+end)
+
+controlCenterTrigger = false
+Trigger.OnEnteredFootprint(ControlCenterTrigger, function(a, id)
+	if not controlCenterTrigger and a.Owner == player and a.Type == "e1" then
+		controlCenterTrigger = true
+		FTurPrisoners.Kill()
+		FTurLeft.Kill()
+		FTurRight.Kill()
+		FTurBottom.Kill()
+		player.MarkCompletedObjective(sovietObjective1)
+	end
+end)
+
+controlCenterEngineerTrigger = false
+Trigger.OnEnteredFootprint(ControlCenterEngineerTrigger, function(a, id)
+	if not controlCenterEngineerTrigger and a.Owner == player and a.Type == "e6" then
+		controlCenterEngineerTrigger = true
+		Actor.Create("ftur", true, { Owner = player, Location = FTur1Goal.Location})
+		Actor.Create("ftur", true, { Owner = player, Location = FTur2Goal.Location})
+		Camera.Position = CameraGoalCenter1.CenterPosition
+		if not cameraGoalRightTrigger then
+			Actor.Create("camera", true, { Owner = player, Location = CameraGoalCenter1.Location })
+			Actor.Create("camera", true, { Owner = player, Location = CameraGoalCenter2.Location })
+			Actor.Create("camera", true, { Owner = player, Location = CameraGoalCenter3.Location })
+		end
+		Utils.Do(GoalGuards, function(actor)
+			actor.Attack(FTur1Goal)
```

Comment:
This wants a `IsDead` check, as it crashes otherwise (if you already killed `GoalGuards`).
```
if not actor.IsDead then
	actor.Attack(FTur1Goal)
end